lib/dynamic_type/test/unary_ops.cpp

```diff
@@ -21,43 +21,67 @@
     EXPECT_EQ((op DoubleInt64BoolVec(2L)).as<decltype(op 2L)>(), (op 2L));    \
     EXPECT_THAT(                                                              \
         [&]() { op DoubleInt64Bool(); },                                      \
-        ::testing::ThrowsMessage<std::runtime_error>(                         \
-            ::testing::HasSubstr("Cannot compute ")));                        \
+        ::testing::ThrowsMessage<std::runtime_error>(::testing::HasSubstr(    \
+            "Result is dynamic but not convertible to result type")));        \
     EXPECT_THAT(                                                              \
         [&]() { op DoubleInt64BoolVec(std::vector<DoubleInt64BoolVec>{}); },  \
-        ::testing::ThrowsMessage<std::runtime_error>(                         \
-            ::testing::HasSubstr("Cannot compute ")));                        \
+        ::testing::ThrowsMessage<std::runtime_error>(::testing::HasSubstr(    \
+            "Result is dynamic but not convertible to result type")));        \
     static_assert(op opcheck<int_or_bool##SomeType>);                         \
     EXPECT_THAT(                                                              \
         [&]() { op int_or_bool##SomeType(SomeType{}); },                      \
-        ::testing::ThrowsMessage<std::runtime_error>(                         \
-            ::testing::HasSubstr("Cannot compute ")));                        \
+        ::testing::ThrowsMessage<std::runtime_error>(::testing::HasSubstr(    \
+            "Result is dynamic but not convertible to result type")));        \
   }
 
 TEST_UNARY_OP(Positive, +, Int);
 TEST_UNARY_OP(Negative, -, Int);
 TEST_UNARY_OP(BinaryNot, ~, Int);
-TEST_UNARY_OP(LogicalNot, !, Bool);
-
 #undef TEST_UNARY_OP
 
+TEST_F(DynamicTypeTest, LogicalNot) {
+  static_assert(!opcheck<DoubleInt64Bool>);
+  static_assert(!opcheck<DoubleInt64BoolVec>);
+  static_assert(std::is_same_v<decltype(!DoubleInt64Bool(2L)), bool>);
```

Comment:
This now returns `bool` because `!double`, `!int64_t` and `!bool` are all bool, so the return value does not have to be dynamic.